src/MetricsTrait.php

```diff
@@ -0,0 +1,143 @@
+<?php
+/*
+ * Copyright 2023 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+namespace Google\Auth;
+
+/**
+ * Provides methods for fetching and updation of auth metrics headers.
+ *
+ * @internal
+ */
+trait MetricsTrait
+{
+    public static $metricsHeaderKey = 'x-goog-api-client';
+
+    // Auth request type
+    public static $requestTypeAccessToken = 'auth-request-type/at';
+    public static $requestTypeIdToken = 'auth-request-type/it';
+    public static $requestTypeMdsPing = 'auth-request-type/mds';
+
+    // Credential type
+    public static $credTypeUser = 'cred-type/u';
+    public static $credTypeSaAssertion = 'cred-type/sa';
+    public static $credTypeSaJwt = 'cred-type/jwt';
+    public static $credTypeSaMds = 'cred-type/mds';
+    public static $credTypeSaImpersonate = 'cred-type/imp';
+
+    // TODO: Find a way to get the auth version
+    // Auth library version
+    public static $version = '10.0.0';
```

Comment:
Why is the placeholder `10.0.0`?

I'd suggest making this a placeholder function instead, like this:

```php
private static getAuthLiberaryVersion(): string
{
    throw new LogicException('This must be implemented by the class which uses the trait');
}
```